src/main/java/seedu/address/model/contact/Contact.java

```diff
@@ -108,8 +108,18 @@ public boolean hasSameFields(Contact otherContact) {
         boolean hasSameNameAndNickname = hasSameNickname && otherContact.getName().equals(getName());
         boolean hasSameNonEmptyNickname =
                 hasSameNickname && !otherContactNickname.isEmpty() && !currentContactNickname.isEmpty();
+        boolean hasPresidentRoleCurrentContact = checkForPresident(getRoles());
+        boolean hasPresidentRoleOtherContact = checkForPresident(otherContact.getRoles());
+        boolean isBothPresident = hasPresidentRoleCurrentContact && hasPresidentRoleOtherContact;
+        return hasSameTelegramHandle || hasSameEmail || hasSameNameAndNickname
+                || hasSameNonEmptyNickname || isBothPresident;
+    }
 
-        return hasSameTelegramHandle || hasSameEmail || hasSameNameAndNickname || hasSameNonEmptyNickname;
+    private boolean checkForPresident(Set<Role> roles) {
+        final int foundStatus = 1;
+        return roles.stream().map(role -> role.roleName)
+                .filter(roleName -> roleName.equalsIgnoreCase(Role.PRESIDENT))
+                .count() == foundStatus;
```

Comment:
It may be better to use anymatch here? Since it is not intuitive to use count when the maximum (should be) 1

Reply:
Oh ya I agree with, thanks for pointing out.